script.js

```diff
@@ -0,0 +1,647 @@
+    row4.classList.remove('active');
+    })
+}
+shiftDigit();
+
+
+
+
+
+
+const row2lang = document.querySelector('.keyboard__row2Eng');
+const row3lang = document.querySelector('.keyboard__row3Eng');
+const row4lang = document.querySelector('.keyboard__row4Eng');
+let key = document.querySelectorAll('.keyboard__key');
+
+document.onkeydown = function (event) {
+    //keyboardNum.push(event.charCode);
+    //console.log(keyboardNum);
+    if (event.key === 'Backspace') {
+        let position = display.selectionStart - 1;
+        if (position >= 1) {
+            let str = display.innerHTML;
+            let res = str.substring(0, position) + '' + str.substring(position + 1);
+            display.innerHTML = res;
+            display.focus();
+        }else {
+            let str = display.innerHTML;
+            let res = str.substring(0, str.length - 1);
+            display.innerHTML = res;
+        }
+    } else if (event.key === 'CapsLock') {
+        if (CAPS2.className === 'keyboard__row2Caps' || row2.className === 'keyboard__row2') {
+            CAPS2.classList.add('active');
+            row2.classList.add('active');
+            CAPS3.classList.add('active');
+            row3.classList.add('active');
+            CAPS4.classList.add('active');
+            row4.classList.add('active');
+        } else if (CAPS3.className === 'keyboard__row3Caps active' || row3.className === 'keyboard__row3 active') {
+            CAPS2.classList.remove('active');
+            row2.classList.remove('active');
+            CAPS3.classList.remove('active');
+            row3.classList.remove('active');
+            CAPS4.classList.remove('active');
+            row4.classList.remove('active');
+        }
+    } else if (event.key === 'Delete') {
+        let position = display.selectionStart;
+            if (position >= 0) {
+                let str = display.innerHTML;
+                let res = str.substring(0, position) + '' + str.substring(position + 1);
+                display.innerHTML = res;
+                display.focus();
+            }
+    } else if (event.key === 'Tab') {
+        let res = display.innerHTML;
+            let position = display.selectionStart - 1;
+            if (position >= 0) {
+                let res1 = res.substring(0, position + 1) + ' '+ ' ' + ' '+ ' ' + res.substring(position + 1);
+                return display.innerHTML = res1;
+            } else {
+                display.innerHTML = res + ' ' + ' ' + ' ' + ' ';
+            }
+    }
+    else if (event.key === 'Enter') {
+        let position = display.selectionStart;
+        if (position >= 1) {
+            let str = display.innerHTML;
+            let res = str.substring(0, position) + '\r\n' + str.substring(position);
+            display.innerHTML = res;
+        }else {
+            let str = display.innerHTML;
+            let res = str + '\r\n';
+            display.innerHTML = res;
+        } 
+    }
+    else if (event.key === 'Shift') {
```

Comment:
all strings like 'Shift' , 'Enter' and so on should be consts. In TS in further tasks you should use enums. So in this way you have no chance to make misprint. Now you can use or object "keys" or "buttons" with appropriate fields.